python/python/lance/dataset.py

```diff
@@ -687,16 +709,18 @@ def from_fragments(fragments: Iterable[LanceFragment], **kwargs):
         -------
         LanceScanner
         """
-        # TODO: can we take kwargs here for the scanner builder?
-        # TODO: peek instead of list
+        warnings.warn(
+            "Deprecated in version 0.4.13. Use the fragments argument in LanceDataset.scanner() instead.",
```

Comment:
we might only have 1-2 users are using this method.  Let's make sure they know, and we can delete this method in a few versions later?

Reply:
Actually this method is `semi-private` method, right? so users dont directly use it, instead they use `Dataset.scanner(fragments=...)`?

Reply:
`Dataset.scanner(fragments=...)` doesn't use this function, so if no one is using `from_fragments()` we can delete it.

Reply:
Sg. Let's delete it then.